examples/WeatherChecker/main.ipynb

```diff
@@ -0,0 +1,569 @@
+        "            <button onclick=\"window.history.back()\">🔙 Go Back</button>\n",
+        "        </div>\n",
+        "    </body>\n",
+        "    </html>\n",
+        "    \"\"\"\n",
+        "    return html\n"
+      ]
+    },
+    {
+      "cell_type": "raw",
+      "metadata": {
+        "vscode": {
+          "languageId": "raw"
+        }
+      },
+      "source": [
+        "First, we declare global variables and handle the input from the WebFormSource.\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "# Global variables\n",
+        "BRNO_LAT = 49.1951\n",
+        "BRNO_LON = 16.6068\n",
+        "\n",
+        "# Using Open-Meteo API (free, no API key required)\n",
+        "WEATHER_API_URL = \"https://api.open-meteo.com/v1/forecast\"\n",
+        "HISTORICAL_API_URL = \"https://archive-api.open-meteo.com/v1/archive\"\n",
+        "\n",
+        "# Input processing\n",
+        "form = event[\"form\"]\n",
+        "parsed_date_str = date(form[\"Year\"], form[\"Month\"], form[\"Day\"]).strftime(\"%Y-%m-%d\")\n",
+        "parsed_date = datetime.strptime(parsed_date_str, \"%Y-%m-%d\")\n",
+        "today_date = datetime.now().date()\n",
+        "request_date = parsed_date.date()\n"
+      ]
+    },
+    {
+      "cell_type": "raw",
+      "metadata": {
+        "vscode": {
+          "languageId": "raw"
+        }
+      },
+      "source": [
+        "Here, we fetch weather data from Open-Meteo API for Brno.\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "weather_data = {}\n",
+        "hourly_data = []\n",
+        "\n",
+        "try:\n",
+        "    # Determine if we need historical data or current/forecast data\n",
+        "    if request_date < today_date:\n",
+        "        # Historical data\n",
+        "        api_url = HISTORICAL_API_URL\n",
+        "    else:\n",
+        "        # Current or forecast data\n",
+        "        api_url = WEATHER_API_URL\n",
+        "    \n",
+        "    # Prepare API request\n",
+        "    params = {\n",
+        "        \"latitude\": BRNO_LAT,\n",
+        "        \"longitude\": BRNO_LON,\n",
+        "        \"start_date\": parsed_date_str,\n",
+        "        \"end_date\": parsed_date_str,\n",
+        "        \"hourly\": [\"temperature_2m\", \"relative_humidity_2m\", \"wind_speed_10m\", \"surface_pressure\"],\n",
+        "        \"daily\": [\"temperature_2m_max\", \"temperature_2m_min\", \"apparent_temperature_max\", \"apparent_temperature_min\"],\n",
+        "        \"timezone\": \"Europe/Prague\"\n",
+        "    }\n",
+        "    \n",
+        "    response = requests.get(api_url, params=params)\n",
```

Comment:
Rather than using requests so we should use `aiohttp` to make an asynchronous request because otherwise I think that the pipeline will block during this a request and other requests that come into this HTTP endpoint will also be blocked.